TrenchBroomTest/raylib_viewer.odin

```
@@ -0,0 +1,471 @@
+	fmt.printf("Loading map file: %s\n", map_file)
+
+	map_result, parse_err := quakemap.load_map_from_file(&loader, map_file)
+	if parse_err != .None {
+		fmt.printf("Error loading map: %v\n", parse_err)
+		os.exit(1)
+	}
+	loaded_map = map_result
+	defer quakemap.map_destroy(&loaded_map)
+
+	fmt.println("Map loaded successfully!")
+
+	// Initialize Raylib
+	rl.InitWindow(SCREEN_WIDTH, SCREEN_HEIGHT, "Quake Map Viewer")
+	defer rl.CloseWindow()
+
+	rl.SetTargetFPS(60)
+	rl.DisableCursor()
+
+	// Setup camera
+	setup_camera()
+
+	// Load textures
+	load_textures()
+
+	// Convert quake meshes to Raylib meshes
+	convert_quake_meshes_to_raylib()
+	defer cleanup_raylib_meshes()
+
+	fmt.println("Starting render loop...")
+
+	// Main game loop
+	for !rl.WindowShouldClose() {
+		update_camera()
+
+		// Render
+		rl.BeginDrawing()
+		rl.ClearBackground(rl.SKYBLUE)
+
+		rl.BeginMode3D(camera)
+
+		// Draw the map
+		draw_map()
+
+		rl.EndMode3D()
+
+		// Draw minimal UI
+		draw_ui()
+
+		rl.EndDrawing()
+	}
+}
+
+setup_camera :: proc() {
+	initial_pos := rl.Vector3{0, 10, 0}
+
+	if len(loaded_map.spawn_points) > 0 {
+		spawn := loaded_map.spawn_points[0]
+		initial_pos = rl.Vector3 {
+			spawn.position.x * MAP_SCALE,
+			spawn.position.z * MAP_SCALE + 1,
+			-spawn.position.y * MAP_SCALE,
+		}
+		fmt.printf("Starting at spawn point: %v (scaled: %v)\n", spawn.position, initial_pos)
+	} else {
+		center := (loaded_map.map_bounds.min + loaded_map.map_bounds.max) * 0.5
+		initial_pos = rl.Vector3 {
+			center.x * MAP_SCALE,
+			center.z * MAP_SCALE + 5,
+			-center.y * MAP_SCALE,
+		}
+		fmt.printf("Starting at map center: %v (scaled: %v)\n", center, initial_pos)
+	}
+
+	camera = rl.Camera3D {
+		position   = initial_pos,
+		target     = rl.Vector3{initial_pos.x, initial_pos.y - 1, initial_pos.z - 1},
+		up         = rl.Vector3{0, 1, 0},
+		fovy       = 75.0,
+		projection = .PERSPECTIVE,
+	}
+
+	camera_yaw = 0.0
+	camera_pitch = 0.0
+}
+
+load_textures :: proc() {
```

This should return an error enum